packages/block-editor/src/components/block-popover/index.js

```diff
@@ -98,10 +156,14 @@ function BlockPopover(
 			contextElement: selectedElement,
 		};
 	}, [
+		popoverDimensionsRecomputeCounter,
+		isZoomOut,
+		parentSectionElement,
+		selectedElement,
 		bottomClientId,
 		lastSelectedElement,
-		selectedElement,
-		popoverDimensionsRecomputeCounter,
+		isSectionSelected,
+		__unstableContentRef,
```

Comment:
I don't really understand the change here. Can you explain why the block popover component needs to be aware of zoom-out...

Reply:
The reason I'm confused is because for me the "BlockPopover" is just a component that gives you the "position" of the block. So I don't see how zoom-out is related. It's not about giving the position of the toolbar, it's about giving the position of the block. It just happens that toolbar is adjacent to the block (in most cases). I think if we need to adapt the position differently in zoom-out, we should either have clear props in this component to change the behavior (for instance a prop that clearly indicates how we want to change the behavior) or do this computation elsewhere.

Reply:
> "BlockPopover" is just a component that gives you the "position" of the block.

I don't know what to say about that, since this does render a whole popover, it's not some utility called to get the position. I do get the concern though.

>  I think if we need to adapt the position differently in zoom-out, we should either have clear props in this component to change the behavior

That is a good alternative sure. But at some point someone will have to "know" about zoom out.

Let's wait on the faith of this toolbar and if it sticks around return to reflect in code the idea that toolbars "just happen" to be adjacent to blocks. This was intended as a bug fix and it does not introduce a big glaring problem IMO just some minor tech debt reflecting the issue with multiple toolbar types.

Do you agree?

Reply:
yes :)